cloud-agent/service/server/src/main/scala/org/hyperledger/identus/presentproof/controller/http/OOBRequestPresentationInput.scala

```diff
@@ -0,0 +1,160 @@
+  AnoncredNonRevokedIntervalV1,
+  AnoncredPresentationRequestV1,
+  AnoncredRequestedAttributeV1,
+  AnoncredRequestedPredicateV1
+}
+import sttp.tapir.{Schema, Validator}
+import sttp.tapir.json.zio.*
+import sttp.tapir.Schema.annotations.{description, encodedExample}
+import zio.json.{DeriveJsonDecoder, DeriveJsonEncoder, JsonDecoder, JsonEncoder}
+import OOBRequestPresentationInput.annotations
+
+//TODO Should I just use RequestPresentationInput and add the optional fields will that cause any confusion
+final case class OOBRequestPresentationInput(
+    @description(annotations.goalcode.description)
+    @encodedExample(annotations.goalcode.example)
+    goalCode: Option[String] = None,
+    @description(annotations.goal.description)
+    @encodedExample(annotations.goal.example)
+    goal: Option[String] = None,
+    @description(annotations.options.description)
+    @encodedExample(annotations.options.example)
+    options: Option[Options] = None,
+    @description(annotations.proofs.description)
+    @encodedExample(annotations.proofs.example)
+    proofs: Seq[ProofRequestAux],
+    @description(annotations.anoncredPresentationRequest.description)
+    @encodedExample(annotations.anoncredPresentationRequest.example)
+    anoncredPresentationRequest: Option[AnoncredPresentationRequestV1],
+    @description(annotations.claims.description)
+    @encodedExample(annotations.claims.example)
+    claims: Option[zio.json.ast.Json.Obj],
+    @description(annotations.credentialFormat.description)
+    @encodedExample(annotations.credentialFormat.example)
+    credentialFormat: Option[String],
+)
+
+object OOBRequestPresentationInput {
+  object annotations {
+    object goalcode
+        extends Annotation[String](
+          description =
+            "A self-attested code the receiver may want to display to the user or use in automatically deciding what to do with the out-of-band message.",
+          example = "present-vp"
+        )
+
+    object goal
+        extends Annotation[String](
+          description =
+            "A self-attested string that the receiver may want to display to the user about the context-specific goal of the out-of-band message.",
+          example = "Request proof of vaccine"
+        )
+
+    object options
+        extends Annotation[Option[Options]](
+          description = "The options to use when creating the proof presentation request (e.g., domain, challenge).",
+          example = None
+        )
+    object proofs
+        extends Annotation[Seq[ProofRequestAux]](
+          description =
+            "The type of proofs requested in the context of this proof presentation request (e.g., VC schema, trusted issuers, etc.)",
+          example = Seq.empty
+        )
+
+    object anoncredPresentationRequest
+        extends Annotation[Option[AnoncredPresentationRequestV1]](
+          description = "Anoncred Presentation Request",
+          example = Some(
```

Comment:
@mineme0110, could you ensure that this example is correctly serialized and encoded in the OAS?